### ISSUE TITLE
Timeline: Items and groups not working together 

### ISSUE THREAD

**Issue by Kyps** [almende/vis#3758]
I'm getting the data for items and groups from mongodb through socket.io and if I want to use them together on the timeline, only groups are visible. Although the date which the timeline shows is correct. Items without the groups work fine. 

```
var container = document.getElementById('timeline');
var groups = new vis.DataSet();
var items = new vis.DataSet();
var options = ({stack: false});
socket2.on('vis_group', function (groupdata){
	groups.add(groupdata);
});
socket3.on('vis', function(itemdata){
	items.add(itemdata);
	var timeline = new vis.Timeline(container,items,groups,options);
});
```
Is there a mistake in this code or is it some kind of anomaly?

**Comment by brichbe** [comment]
Without seeing more of your code, the obvious question is:   are you certain socket2 is receiving the group data prior to socket3 receiving your item data?  I'm not familiar with socket.io, but I assume those callbacks are asynchronous and therefore the order is not guaranteed.

**Comment by Kyps** [comment]
Yes they are asynchronous but is the load order important? Do the items have to be loaded in first?

**Comment by brichbe** [comment]
No I don't believe the items would _need_ to be first, and you're right the order _shouldn't_ be the problem. But like I said, without seeing more of your code, and without you giving us more details/an example, it's difficult to help track down your problem.  One possible problem is related to scope and the way you're declaring and creating the timeline inside your socket callback. Try creating the timeline not inside a callback but when the DataSets and the options are first created, then use the socket callbacks to only add the data to those DataSets.
